packages/components/src/border-control/border-control/component.tsx

```diff
@@ -16,19 +16,19 @@ import { VisuallyHidden } from '../../visually-hidden';
 import { contextConnect, WordPressComponentProps } from '../../ui/context';
 import { useBorderControl } from './hook';
 
-import type { BorderControlProps, LabelProps } from '../types';
+import type { BorderControlProps, BorderLabelProps } from '../types';
 
-const BorderLabel = ( props: LabelProps ) => {
-	const { label, hideLabelFromVision } = props;
+const BorderLabel = ( props: BorderLabelProps ) => {
+	const { label, hideLabelFromVision, ...labelProps } = props;
 
 	if ( ! label ) {
 		return null;
 	}
 
 	return hideLabelFromVision ? (
-		<VisuallyHidden as="legend">{ label }</VisuallyHidden>
+		<VisuallyHidden { ...labelProps }>{ label }</VisuallyHidden>
 	) : (
-		<StyledLabel as="legend">{ label }</StyledLabel>
+		<StyledLabel { ...labelProps }>{ label }</StyledLabel>
```

Comment:
To my understanding, this will render a stray `<label>` element, ie. a `<label>` that isn't associated with any field.